### DISCUSSION TITLE
Typescript ExifDateTime

### DISCUSSION THREAD

**Discussion by captDaylight** [photostructure/exiftool-vendored.js#111] [Q&A] [answered]
I just started using exiftool-vendored, really incredible work.

I'm having some typescript issues after reading my first file. I'm getting `CreateDate` object successfully as `ExifDateTime` but when I try to call the method `toDate` my editor is saying it doesn't exist. 

<img width="740" alt="Screen Shot 2022-05-24 at 5 03 45 PM" src="https://user-images.githubusercontent.com/700617/170131994-5b36bfd8-931f-4659-abf4-fe6141116315.png">

On the other hand, the method `toString` doesn't throw an error. Everything works correctly when I use `ts-ignore`.

Any idea why this method is not being recognized?

I'm on Mac OS 10.15.6, Node 16
 - OS and version: [e.g. Mac OS X 10.11.6]
 - Node.JS version [must be an active, current, or maintenance version]

**Comment by mceachen** [comment, marked as answer]
> Any idea why this method is not being recognized?

Please check the typings: `CreateDate` could be:

1. `undefined`
2. an instance of `string`, if the date is encoded with an invalid value (some manufacturers seem to enjoy the date "0000:00:00")
3. an instance of `ExifDateTime`

You'll need to verify that `CreateDate` is an `instanceof ExifDateTime` before calling methods on that interface are allowed.
